### PR TITLE
Add UUID to yaml parsing

### PR DESCRIPTION
**Pull request by kalinon**
Pretty self explanatory, adds the ability to parse UUID fields from strings.

### CODE REVIEW

File: src/uuid/yaml.cr [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  if node.is_a?(YAML::Nodes::Scalar)
+    value = node.value
+    ctx.record_anchor(node, value)
+    UUID.new(value)
+  else
+    node.raise "Expected String, not #{node.kind}"
+  end
+end
+
+struct UUID
```
Comment (straight-shoota) [inline src/uuid/yaml.cr:36]: Since `UUID` needs to be reopened anyways, you could move the constructor method into the namespace as well.

**Review by straight-shoota — APPROVED** [review]

File: src/uuid/yaml.cr (hunk 1 above)
Comment (kalinon) [reply src/uuid/yaml.cr]: Done!

File: src/uuid/yaml.cr [hunk 2]
```diff
@@ -0,0 +1,48 @@
+require "yaml"
+require "uuid"
+
+struct UUID
+  # Creates UUID from YAML using `YAML::ParseContext`.
```
Comment (Sija) [inline src/uuid/yaml.cr:5]:
```suggestion
  # Creates `UUID` from YAML using `YAML::ParseContext`.
```

File: src/uuid/yaml.cr [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    if node.is_a?(YAML::Nodes::Scalar)
+      value = node.value
+      ctx.record_anchor(node, value)
+      UUID.new(value)
+    else
+      node.raise "Expected String, not #{node.kind}"
+    end
+  end
+
+  # Returns UUID as YAML value.
```
Comment (Sija) [inline src/uuid/yaml.cr:37]:
```suggestion
  # Returns `UUID` as YAML value.
```

**Review by straight-shoota — APPROVED** [review]

**Review by asterite — APPROVED** [review]

**Pull request merged**